### PR TITLE
feat: replace ChainTypes to SupportedChainIds in Swapper

### PR DESCRIPTION
**Pull request by elmutt**
**NOTE: This is a breaking change**

- Replace `ChainTypes` with `SupportedChainIds`
- We are in the process of removing the `ChainTypes` types throughout web. This PR removes it from Swapper specifically.

### CODE REVIEW

File: packages/swapper/src/api.ts [hunk 1]
```diff
@@ -3,20 +3,36 @@ import { HDWallet } from '@shapeshiftoss/hdwallet-core'
...
   SwapperType
 } from '@shapeshiftoss/types'
+import { ethereum } from '@shapeshiftoss/types/src/chain-adapters'
+import { ChainSpecific } from '@shapeshiftoss/types/src/utility'
+
 export type SupportedAssetInput = {
   assetIds: CAIP19[]
 }
 
+export enum ChainIdTypes {
```
Comment (DaoDev44) [inline packages/swapper/src/api.ts:18]: move to types

File: packages/swapper/src/swappercli.ts [hunk 2]
```diff
@@ -1,140 +1,140 @@
...
-    mnemonic: MNEMONIC,
-    deviceId: DEVICE_ID
-  }
-  const wallet = new NativeHDWallet(nativeAdapterArgs)
-  await wallet.initialize()
-
-  return wallet
-}
+// import { AssetService } from '@shapeshiftoss/asset-service'
+// import { ChainAdapterManager } from '@shapeshiftoss/chain-adapters'
```
Comment (DaoDev44) [inline packages/swapper/src/swappercli.ts:2]: fix this

File: packages/swapper/src/swappers/zrx/ZrxApprovalNeeded/ZrxApprovalNeeded.ts [hunk 3]
```diff
@@ -9,21 +9,17 @@ import { ZrxSwapperDeps } from '../ZrxSwapper'
...
-  { quote, wallet }: ApprovalNeededInput<ChainTypes>
+  { quote, wallet }: ApprovalNeededInput<ChainIdTypes>
 ): Promise<ApprovalNeededOutput> {
   const { sellAsset } = quote
 
   if (sellAsset.symbol === 'ETH') {
     return { approvalNeeded: false }
   }
 
-  if (sellAsset.chain !== ChainTypes.Ethereum) {
```
Comment (DaoDev44) [inline packages/swapper/src/swappers/zrx/ZrxApprovalNeeded/ZrxApprovalNeeded.ts:20]: add this back in

File: packages/swapper/src/swappers/zrx/utils/test-data/setupZrxDeps.ts [hunk 4]
```diff
@@ -1,5 +1,6 @@
 import { ChainAdapterManager } from '@shapeshiftoss/chain-adapters'
 import { chainAdapters, ChainTypes } from '@shapeshiftoss/types'
+// import { ChainIdTypes } from '../../../../api'
```
Comment (DaoDev44) [inline packages/swapper/src/swappers/zrx/utils/test-data/setupZrxDeps.ts:3]: remove

File: packages/swapper/src/swappers/zrx/zrxBuildTrade/zrxBuildTrade.ts [hunk 5]
```diff
@@ -58,12 +57,6 @@ export async function zrxBuildTrade(
     )
   }
 
-  if (buyAsset.chain !== ChainTypes.Ethereum) {
-    throw new SwapError(
```
Comment (DaoDev44) [inline packages/swapper/src/swappers/zrx/zrxBuildTrade/zrxBuildTrade.ts:62]: add back

File: packages/swapper/src/api.ts (hunk 1 above)
Comment (elmutt) [reply packages/swapper/src/api.ts:18]: rename to supportedChainIds

File: packages/swapper/src/api.ts [hunk 6]
```diff
@@ -3,20 +3,36 @@ import { HDWallet } from '@shapeshiftoss/hdwallet-core'
...
+  Ethereum = 'eip155:1/slip44:60',
+  Bitcoin = 'bip122:000000000019d6689c085ae165831e93/slip44:0',
+  Cosmos = 'cosmos:cosmoshub-4/slip44:118',
+  Osmosis = 'cosmos:osmosis-1/slip44:118'
```
Comment (elmutt) [inline packages/swapper/src/api.ts:22]: These strings are caip19. we want caip2

File: packages/swapper/src/api.ts [hunk 7]
```diff
@@ -4,17 +4,32 @@ import {
...
+  SupportedChainIds,
   SwapperType
 } from '@shapeshiftoss/types'
+import {} from '@shapeshiftoss/types/src/utility'
+
 export type SupportedAssetInput = {
   assetIds: AssetId[]
 }
 
+// TODO(ryankk): ChainSpecificQuoteFeeData and QuoteFeeData is being used as a stopgap until removal of ChainTypes
```
[the remark this answers is not in the document]
Comment (kaladinlight) [reply packages/swapper/src/api.ts:19]: Need to confirm this works through the whole stack. Build is at least happy

File: packages/types/src/chain-adapters/index.ts [hunk 8]
```diff
@@ -6,7 +6,7 @@ import * as bitcoin from './bitcoin'
 import * as cosmos from './cosmos'
 import * as ethereum from './ethereum'
 import * as osmosis from './osmosis'
-export { bitcoin, cosmos, ethereum }
+export { bitcoin, cosmos, ethereum, ChainSpecific }
```
Comment (kaladinlight) [inline packages/types/src/chain-adapters/index.ts:9]: Remove `ChainSpecific` export here

**Review by kaladinlight — APPROVED** [review]

**Review by DaoDev44 — APPROVED** [review]

**Pull request merged**